### PR TITLE
Improve past gen item and move viability

### PR DESCRIPTION
**Pull request by petuuuhhh**
Improve berry search/Hidden Power search display for past gens. I know Zarel you wanted me to edit my original pull req initially, then said you could do it, but here it is done for you.

### CODE REVIEW

File: src/battle-dex-search.ts [hunk 1]
```diff
@@ -1177,6 +1177,8 @@ class BattleMoveSearch extends BattleTypedSearch<'move'> {
 			return ['technician', 'serenegrace'].includes(abilityid) || !moves.includes('powergem');
 		case 'aurawheel':
 			return species.baseSpecies === 'Morpeko';
+		case 'aurorabeam':
+			return !moves.includes('icebeam');
```
Comment (Zarel) [inline src/battle-dex-search.ts:1181]: > /ds aurorabeam, !icebeam:
> No Pokémon found.

```suggestion
```

I'd rather not.

Comment (urkerab) [reply src/battle-dex-search.ts:1181]: Xerneas only gets Ice Beam in Gen 7.

Comment (KrisXV) [reply src/battle-dex-search.ts:1181]: I think you mean Aurora Beam.

Comment (urkerab) [reply src/battle-dex-search.ts:1181]: It got Aurora Beam in Gen 6... I'm not saying that it's viable though.

Comment (Zarel) [reply src/battle-dex-search.ts:1181]: Does Gen 7 Xerneas actually hit this code path, though? I suppose this is worth it if so.

Comment (petuuuhhh) [reply src/battle-dex-search.ts:1181]: urkerab is saying Gen 6 Xerneas hits the path. That's why it's Xerneas tech in Gen 6.

Comment (Zarel) [reply src/battle-dex-search.ts:1181]: Works for me. Thanks for the changes!

**Pull request merged**